src/emqx_misc.erl

```diff
@@ -529,6 +530,19 @@ redact(K, V) ->
 redact_v(V) when is_binary(V) -> <<?REDACT_VAL>>;
 redact_v(_V) -> ?REDACT_VAL.
 
+force_relative_filename_join([H | T]) ->
+    filename:join([H | lists:map(fun ensure_filename_to_relative/1, T)]).
+
+ensure_filename_to_relative(<<$/,Path/binary>> = RawPath) ->
+    ?LOG(warning, "the parts of the filename path should be relative, invalid path: ~p", [RawPath]),
```

Comment:
I think it's better to check the paths during config validation and report schema validation error than to try to fix it in the runtime.
Validation can be done with `filename:pathtype` or https://www.erlang.org/doc/man/filelib.html#safe_relative_path-2

Reply:
I found a bug in the `filename:safe_relative_path` 
https://github.com/erlang/otp/issues/6460
Maybe I will pause this PR for a while